src/rtkcmn.c

```diff
@@ -1020,6 +1022,24 @@ extern double *eye(int n)
     if ((p=zeros(n,n))) for (i=0;i<n;i++) p[i+i*n]=1.0;
     return p;
 }
+
+/* dot product -----------------------------------------------------------------
+ * inner product of vectors of size 2
+ * args   : double *a,*b     I   vectors a and b
+ * return : a'*b
+ *-----------------------------------------------------------------------------*/
+extern double dot2(const double* a, const double* b) { return a[0] * b[0] + a[1] * b[1]; }
```

Comment:
It appears that the dot product patch could be split out into a separate PR, perhaps just a case of creating a new branch off demo5 and a git cherry-pick.

Reply:
Done! That's #196 now!